fastar/update_rules.py

```diff
@@ -1,17 +1,18 @@
+import jax.lax as lax
+import jax.numpy as np
 import numpy as onp
-
 from jax import vjp
-import jax.numpy as np
-from jax.util import curry, safe_zip, safe_map, unzip2
-import jax.lax as lax
 from jax.abstract_arrays import make_shaped_array
+from jax.util import curry, safe_zip, safe_map, unzip2
 
 import fastar.interpreter as fa
 import fastar.primitives as primitives
 import fastar.util as util
 
 map = safe_map
 zip = safe_zip
+all = slice(None, None)
```

Comment:
I think over-riding builtins should be done with care. The above two over-rides at least replace with something which is very similar.

This over-ride could confuse people reading the code quite a bit because this all has a very different meaning to the built in.

It’s annoying that there isn’t already a shorthand for this ‘empty’ slice. Could maybe call it noneslice?

Reply:
Good point, override is an accident. Renaming to noneslice.